share/eds/byzantine/bad_encoding.go

```diff
@@ -141,6 +141,20 @@ func (p *BadEncodingProof) Validate(hdr libhead.Header) error {
 		root = merkleColRoots[p.Index]
 	}
 
+	// find at least 1 non-empty share
+	nonEmpty := false
+	for _, share := range p.Shares {
+		if share == nil {
+			continue
+		}
+		nonEmpty = true
+		break
+	}
+
+	if !nonEmpty {
+		return errors.New("fraud: invalid proof: no shares provided to reconstruct row/col")
+	}
+
```

Comment:
We can simply check len of `shares` right after the loop below for the same effect.

Reply:
not exactly, because we can receive a slice with all nil values inside.

Reply:
oh, I got you. Thanks

Reply:
You can preallocate by cap then. Then it's much less code

Reply:
let's at least make it not double negation, so the check should be `empty := true` and then check if `empty`

Reply:
done